Sources/Turf/Geometries/LineString.swift

```diff
@@ -215,4 +215,114 @@ extension LineString {
+        let dx = coordinate1.longitude - coordinate2.longitude
+        let dy = coordinate1.latitude - coordinate2.latitude
+        return dx * dx + dy * dy
+  }
+
+    private func simplifyRadialDistance(_ coordinates: [CLLocationCoordinate2D], tolerance: Double) -> [CLLocationCoordinate2D] {
+        guard coordinates.count > 2 else { return coordinates }
+
+        var prevCoordinate = coordinates[0]
+        var newCoordinates = [prevCoordinate]
+        var coordinate = coordinates[1]
+
+        for index in 1 ..< coordinates.count {
+            coordinate = coordinates[index]
+
+            if getSquareDistance(coordinate1: coordinate, coordinate2: prevCoordinate) > tolerance {
+                newCoordinates.append(coordinate)
+                prevCoordinate = coordinate
+            }
+        }
+
+        if prevCoordinate != coordinate {
+            newCoordinates.append(coordinate)
+        }
+
+        return newCoordinates
+    }
+
+    private func getSquareSegmentDistance(_ coordinate: CLLocationCoordinate2D, segmentStart: CLLocationCoordinate2D, segmentEnd: CLLocationCoordinate2D) -> CLLocationDistance {
+
+        var x = segmentStart.latitude
+        var y = segmentStart.longitude
+        var dx = segmentEnd.latitude - x
+        var dy = segmentEnd.longitude - y
+
+        if dx != 0 || dy != 0 {
+            let t = ((segmentStart.latitude - x) * dx + (coordinate.longitude - y) * dy) / (dx * dx + dy * dy)
+            if t > 1 {
+                x = segmentEnd.latitude
+                y = segmentEnd.longitude
+            } else if t > 0 {
+                x += dx * t
+                y += dy * t
+            }
+        }
+
+        dx = coordinate.latitude - x
+        dy = coordinate.longitude - y
+
+        return dx * dx + dy * dy
+    }
+
+    private func simplifyDouglasPeuckerStep(_ coordinates: [CLLocationCoordinate2D], first: Int, last: Int, tolerance: Double, simplified: inout [CLLocationCoordinate2D]) {
+
+        var maxSquareDistance = tolerance
+        var index = 0
+
+        for i in first + 1 ..< last {
+            let squareDistance = getSquareSegmentDistance(coordinates[i], segmentStart: coordinates[first], segmentEnd: coordinates[last])
+
+            if squareDistance > maxSquareDistance {
+                index = i
+                maxSquareDistance = squareDistance
+            }
+        }
+
+        if maxSquareDistance > tolerance {
+            if index - first > 1 {
+                simplifyDouglasPeuckerStep(coordinates, first: first, last: index, tolerance: tolerance, simplified: &simplified)
+            }
+            simplified.append(coordinates[index])
+            if last - index > 1 {
+                simplifyDouglasPeuckerStep(coordinates, first: index, last: last, tolerance: tolerance, simplified: &simplified)
+            }
+        }
+    }
+
+    private func simplifyDouglasPeucker(_ coordinates: [CLLocationCoordinate2D], tolerance: Double!) -> [CLLocationCoordinate2D] {
```

Comment:
Does `tolerance` need to be an implicitly unwrapped optional? We can avoid the risk of a crash by making it a `Double`, since `simplifyDouglasPeuckerStep(_:first:last:tolerance:simplified:)` takes a `Double`.

Reply:
Corrected.